components/create_comment/create_comment.jsx

```diff
@@ -430,12 +430,25 @@ class CreateComment extends React.PureComponent {
         e.preventDefault();
         this.updatePreview(false);
 
-        const membersCount = this.props.channelMembersCount;
-        const notificationsToChannel = this.props.enableConfirmNotificationsToChannel && this.props.useChannelMentions;
-        if (notificationsToChannel &&
-            membersCount > Constants.NOTIFY_ALL_MEMBERS &&
+        const {channelMembersCount, enableConfirmNotificationsToChannel, useChannelMentions, isTimezoneEnabled} = this.props;
+        const {draft} = this.state;
+        if (!useChannelMentions && containsAtChannel(draft.message, {checkAllMentions: true})) {
+            const updatedDraft = {
+                ...draft,
+                props: {
+                    ...draft.props,
+                    mentionHighlightDisabled: true,
```

Comment:
Can we trust the client to set this correctly? What if a user that can't do channel-wide mentions were to use the API to manually send a post without this field? It feels to me like this is something that the server would set.

Reply:
The server pr (mattermost/mattermost-server#13902) also ensures the props are set if the user is lacking permission to channel mention.

The reason also have it here is for the split second immediately after creating the post but before the post is actually validated on the server. During that time the message would show up as highlighted (for the user sending the post) if the client did not also set the prop.